### ISSUE TITLE
Fix highlightOnPlot

### ISSUE THREAD

**Issue by lmsimp** [lgatto/pRoloc#79]
```r
library('pRoloc')
library('pRolocGUI')

library('pRolocdata')
data("hyperLOPIT2015ms3r1")
data("hyperLOPIT2015ms3r2")

sampleNames(hyperLOPIT2015ms3r1)
sampleNames(hyperLOPIT2015ms3r2)

pRolocVis(hyperLOPIT2015ms3r1)
prots <- c("Q8VCQ6", "Q99J21")
inds <- match(prots, featureNames(hyperLOPIT2015ms3r1))
fData(hyperLOPIT2015ms3r1)$markers[inds] <- "xxx"
inds2 <- match(prots, featureNames(hyperLOPIT2015ms3r2))
fData(hyperLOPIT2015ms3r2)$markers[inds2] <- "xxx"

x <- MSnSetList(list(hyperLOPIT2015ms3r1, hyperLOPIT2015ms3r2))
```

```r
## Issue 1: mirrorX = TRUE does not work with highlightOnPlot
plot2D(x[[1]])
highlightOnPlot(x[[1]], prots, col = "black", lwd = 2)
plot2D(x[[1]], mirrorX = TRUE)
highlightOnPlot(x[[1]], prots , col = "black", lwd = 2, mirrorX = TRUE)
```

```r
## Issue 2: datasets in remapped list are different sizes and no option to pass 
## method = "none" to highlightOnPlot (so essentially we do another PCA on a 
## PCA dataspace and end up with the wrong coords)

## remap data
r <- pRoloc:::remap(x, ref = 1)
dim(r[[1]])
dim(r[[2]])

plot2D(x[[1]])
highlightOnPlot(x[[1]], prots , col = "black", lwd = 2)

## plot reference dataset (no re-mapping) and use method = "none"
plot2D(r[[1]], method = "none")
highlightOnPlot(r[[1]], prots, col = "black", lwd = 2, method = "none")
```

**Comment by lgatto** [comment]
Regarding issue 2:

```r
> sapply(msnsets(x), dim)
     [,1] [,2]
[1,] 5489 6268
[2,]   10   10
> 
> r <- pRoloc:::remap(x, ref = 1)
> sapply(msnsets(r), dim)
     [,1] [,2]
[1,] 5489 6268
[2,]   10   10
```

I think this is correct. `remap` does not change the dimensions because remapping in the reference's PCA space does not require the number of rows to be the same; it actually does not expect the features to be the same. To get this behaviour, one would need to first use `commonFeatureNames`.

```r
> x2 <- commonFeatureNames(x)
5032 features in common
> r2 <- pRoloc:::remap(x2, ref = 1)
> sapply(msnsets(x2), dim)
     [,1] [,2]
[1,] 5032 5032
[2,]   10   10
> sapply(msnsets(r2), dim)
     [,1] [,2]
[1,] 5032 5032
[2,]   10   10
```

**Comment by lgatto** [comment]
Back to issue number 2:

```r
plot2D(r[[1]], method = "none")
highlightOnPlot(exprs(r[[1]]), prots, col = "black", lwd = 2)
```
![tmpfig12343243143](https://cloud.githubusercontent.com/assets/384198/15695326/e96f4960-279d-11e6-8dd7-8f5861050a3e.png)

The key here is to pass the matrix, rather than the `MSnSet`. From `?highlightOnPlot`

>  object: The main dataset described as an ‘MSnSet’ or a ‘matrix’ with
>          the coordinates of the features on the PCA plot produced (and
>          invisibly returned) by ‘plot2D’.

Does that solve or clarify issue 2?

**Comment by lgatto** [comment]
Issue 1 is now fixed on github (version 1.13.3). Will push to Bioc and backport. 

Closing the issue now. @lmsimp - please re-open if necessary.
